BitmovinYoSpaceModule/Classes/BitmovinYospacePlayer.swift

```diff
@@ -919,43 +918,29 @@ extension BitmovinYospacePlayer: PlayerListener {
         liveAdPaused = isLive && isAd
         yospacesession?.playerEventDidOccur(YOPlayerEvent.playbackPauseEvent, playhead: currentTimeWithAds())
 
-        for listener: PlayerListener in listeners {
-            listener.onPaused?(PausedEvent(time: currentTime), player: player)
-        }
+        let pausedEvent = PausedEvent(time: currentTime)
+        emitPreprocessedEvent(event: pausedEvent, player: player)
     }
 
     public func onSourceUnloaded(_ event: SourceUnloadedEvent, player: Player) {
         if sessionStatus != .notInitialised {
             // the yospace sessionManager.shutdown() call is asynchronous. If the user just calls `load()` on second playback without calling `unload()` we end up canceling both the old session and the new session. This if statement keeps track of that
             resetYospaceSession()
         }
-        for listener: PlayerListener in listeners {
-            listener.onSourceUnloaded?(event, player: player)
-        }
+
+        emitPreprocessedEvent(event: event, player: player)
     }
 
     public func onStallStarted(_ event: StallStartedEvent, player: Player) {
         yospacesession?.playerEventDidOccur(YOPlayerEvent.playbackStallEvent, playhead: currentTimeWithAds())
 
-        for listener: PlayerListener in listeners {
-            listener.onStallStarted?(event, player: player)
-        }
+        emitPreprocessedEvent(event: event, player: player)
     }
```

Comment:
Those events are not really pre-processed but for the simplicity on this PR I kept this as is.